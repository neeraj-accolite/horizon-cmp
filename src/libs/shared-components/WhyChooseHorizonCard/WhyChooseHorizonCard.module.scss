.card {
  border-radius: 0.75rem; // 12px
  padding: 1rem 0.75rem; // 16px 12px
  width: 19.75rem; // 316px
  height: 11.75rem; // 188px
  background-color: white;
}

.title {
  text-align: start;
  margin-bottom: 0.5rem;
  font-size: 1.5rem; // 24px
  font-weight: 600;
}

.list {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  list-style: disc;
  padding-left: 1.5rem;
  text-align: left;
}
