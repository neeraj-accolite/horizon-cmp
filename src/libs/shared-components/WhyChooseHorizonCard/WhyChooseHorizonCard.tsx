import React from 'react';
import { Card } from '@shared-components';
import styles from './WhyChooseHorizonCard.module.scss';

export interface WhyChooseHorizonCardProps {
  title: string;
  listItems: string[];
  style?: React.CSSProperties;
}

const WhyChooseHorizonCard: React.FC<WhyChooseHorizonCardProps> = ({
  title,
  listItems,
  style = {},
}) => {
  return (
    <Card
      style={{
        borderRadius: '12px',
        padding: '16px 12px',
        width: '316px',
        height: '188px',
        backgroundColor: 'white',
        ...style,
      }}
    >
      <div className={`${styles.title}`}>
        <h2>{title}</h2>
      </div>
      <ul className={`${styles.list}`}>
        {listItems.map((item, index) => (
          <li key={index}>{item}</li>
        ))}
      </ul>
    </Card>
  );
};

export default WhyChooseHorizonCard;
