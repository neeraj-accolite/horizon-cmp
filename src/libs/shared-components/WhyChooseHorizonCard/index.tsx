export { default as WhyChooseHorizonCard } from './WhyChooseHorizonCard';
