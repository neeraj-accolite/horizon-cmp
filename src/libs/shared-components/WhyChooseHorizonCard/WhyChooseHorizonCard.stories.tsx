import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { WhyChooseHorizonCardProps } from './WhyChooseHorizonCard';
import { WhyChooseHorizonCard } from '@shared-components';
export default {
  title: 'Components/WhyChooseHorizonCard',
  component: WhyChooseHorizonCard,
} as Meta;

const Template: StoryFn<WhyChooseHorizonCardProps> = (args) => (
  <div
    style={{
      display: 'flex',
      justifyContent: 'center',
      padding: '20px',
      backgroundColor: '#f9f9f9',
    }}
  >
    <WhyChooseHorizonCard {...args} />
  </div>
);

export const Default = Template.bind({});
Default.args = {
  title: 'Why Choose Horizon?',
  listItems: [
    'Same-day service & next-day installation',
    '24-hour availability',
    'On-time guarantee',
  ],
};

export const CustomStyles = Template.bind({});
CustomStyles.args = {
  title: 'Custom Horizon Card',
  listItems: [
    'Custom style applied',
    'Different padding',
    'Unique color scheme',
  ],
};
