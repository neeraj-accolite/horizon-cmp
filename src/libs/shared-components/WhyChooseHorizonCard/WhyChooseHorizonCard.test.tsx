import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { WhyChooseHorizonCardProps } from './WhyChooseHorizonCard';
import { WhyChooseHorizonCard } from '@shared-components';
describe('WhyChooseHorizonCard', () => {
  const defaultProps: WhyChooseHorizonCardProps = {
    title: 'Why Choose Horizon?',
    listItems: [
      'Same-day service & next-day installation',
      '24-hour availability',
      'On-time guarantee',
    ],
  };

  it('renders the card title correctly', () => {
    render(<WhyChooseHorizonCard {...defaultProps} />);
    const titleElement = screen.getByRole('heading', {
      name: /Why Choose Horizon\?/i,
    });
    expect(titleElement).toBeInTheDocument();
  });

  it('renders all list items provided in props', () => {
    render(<WhyChooseHorizonCard {...defaultProps} />);
    defaultProps.listItems.forEach((item) => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });
  });

  it('renders an empty list when no listItems are provided', () => {
    render(<WhyChooseHorizonCard title="Empty List Test" listItems={[]} />);
    const listElement = screen.getByRole('list');
    expect(listElement.children.length).toBe(0);
  });

  it('matches the snapshot', () => {
    const { container } = render(<WhyChooseHorizonCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
