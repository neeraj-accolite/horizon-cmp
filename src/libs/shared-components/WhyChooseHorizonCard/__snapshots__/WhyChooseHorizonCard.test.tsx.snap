// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhyChooseHorizonCard matches the snapshot 1`] = `
<div>
  <div
    style="border-radius: 12px; padding: 16px 12px; width: 316px; height: 188px; background-color: white;"
  >
    <div
      class="title"
    >
      <h2>
        Why Choose Horizon?
      </h2>
    </div>
    <ul
      class="list"
    >
      <li>
        Same-day service & next-day installation
      </li>
      <li>
        24-hour availability
      </li>
      <li>
        On-time guarantee
      </li>
    </ul>
  </div>
</div>
`;
